utils/src/main/java/dev/arkbuilders/components/utils/PathExt.kt

```diff
@@ -0,0 +1,18 @@
+package dev.arkbuilders.components.utils
+
+import java.nio.file.Path
+
+fun Path.isParentOrSame(path: Path): Boolean {
+    return path.toString().startsWith(this.toString())
+}
+
+fun Path.isChildOrSame(path: Path): Boolean {
+    return this.toString().startsWith(path.toString())
+}
+
+fun Path.hasNestedOrParentalRoot(roots: Set<Path>): Boolean {
```

Comment:
We can accept `Iterable<Path>`
The set sounds logical, we can't have several identical roots, but we usually store them in `List`